components/automate-ui/src/app/pages/team-add-users/team-add-users.component.ts

```diff
@@ -0,0 +1,229 @@
+              this.store.dispatch(new GetTeam({ id }));
+            }
+          });
+      });
+
+      combineLatest(
+        this.store.select(allUsers),
+        this.store.select(userStatus),
+        this.store.select(teamUsers),
+        this.store.select(getTeamUsersStatus))
+        .pipe(
+          map(([users, uStatus, tUsers, tStatus]) => {
+            if (uStatus !== EntityStatus.loadingSuccess ||
+              tStatus !== EntityStatus.loadingSuccess) {
+              this.loading = true;
+              return [];
+            }
+            const sortedUser = users.sort(
+              (a, b) => {
+                // See https://stackoverflow.com/a/38641281 for these options
+                const opts = { numeric: true, sensitivity: 'base' };
+                // sort by name then by id
+                return a.name.localeCompare(b.name, undefined, opts) ||
+                  a.name.localeCompare(b.name, undefined, { numeric: true }) ||
+                  a.id.localeCompare(b.id, undefined, opts);
+                });
+            this.users = sortedUser;
+            // Map UUID membership to user records and remove any entries that don't
+            // map to user records.
+            return at(tUsers, keyBy('membership_id', users))
+              .filter(userRecord => userRecord !== undefined);
+          }),
+          map((users: User[]) => {
+            this.mapOfUsersToFilter = userArrayToHash(users);
+            this.loading = false;
+          }),
+          takeUntil(this.isDestroyed)
+        ).subscribe();
+ }
+
+  ngOnDestroy() {
+    this.isDestroyed.next(true);
+    this.isDestroyed.complete();
+  }
+
+  addUsers(): void {
+    this.addingUsers = true;
+    this.addUsersFailed = '';
+    const userIDs = Object.values(this.usersToAdd).map((user: User) => user.membership_id);
+    this.store.dispatch(new AddTeamUsers(<TeamUserMgmtPayload>{
+      id: this.teamId,
+      user_ids: userIDs
+    }));
+
+    const pendingAdd = new Subject<boolean>();
+    this.store.select(addUsersStatus).pipe(
+      filter(identity),
+      takeUntil(pendingAdd))
+      .subscribe((state) => {
+        if (state === EntityStatus.loadingSuccess) {
+          pendingAdd.next(true);
+          pendingAdd.complete();
+          this.addingUsers = false;
+          this.closePage();
+        }
+        if (state === EntityStatus.loadingFailure) {
+          this.store.select(addTeamUsersStatusHTTPError).pipe(
+            filter(identity),
+            takeUntil(pendingAdd)).subscribe((error: HttpErrorResponse) => {
+              if (error.message === undefined) {
+                this.addUsersFailed = 'An error occurred while attempting ' +
+                'to add users. Please try again.';
+              } else {
+                this.addUsersFailed = `Failed to add users: ${error.message}`;
+              }
+              this.addingUsers = false;
+          });
+        }
+      });
+  }
+
+  public getHeading(): string {
```

Comment:
Majority of your public functions do not have the `public` qualifier explicitly, so should remove it here and on L180. Either way is fine, but should be consistent throughout the file. (And, in a perfect world, throughout the project, but that's for another time, I suppose. 🤷‍♂ )